### PR TITLE
C&DH: Telemetry revamp, ADCS, OBDH, Logging

### PR DESCRIPTION
**Pull request by Ibrassow**
PR changes
- New tasks (CDH, ADCS, EPS, THERMAL, GPS, etc) 
- New telemetry packet following telemetry definitions 
- ADCS: tasks, environmental models, coarse attitude determination
- Optimizations within DH 
- Improvement of constants definition and caching
- Added logging and improvement task debug 
- Preparation for mainboard V1

### CODE REVIEW

**Comment by Ibrassow** [comment]
Hardware tests done.

File: flight/apps/cdh.py
Comment (tomyyyD) [inline flight/apps/cdh.py:1]: Empty file?

File: flight/apps/telemetry/packing.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+    pack_unsigned_long_int,
+    pack_unsigned_short_int,
+    unpack_unsigned_short_int,
+)
+from core import logger
+from core.data_handler import DataHandler as DH
+from micropython import const
+
+
+class TelemetryPacker:
```
Comment (tomyyyD) [inline flight/apps/telemetry/packing.py:40]: Kind of no point in making this a class if every single method has an `@classmethod` decorator

File: flight/apps/comms/rf_mcu.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+class RF_MCU:
+    """
+    High-level interface to communicate with the RF MCU managing the SEMTECH SX1262 RF LoRa module (E22-900M30S).
+    """
+
+    SPI = None
+
+    @classmethod
```
Comment (tomyyyD) [inline flight/apps/comms/rf_mcu.py:20]: Why are these `@classmethods`? Shouldn't the satellite initialize one instance of RF_MCU and that instance will use these methods?

File: flight/hal/drivers/adm1176.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 CircuitPython driver for the adm1176 hot swap controller and I2C power monitor
 
-* Author(s): Max Holliday, Harry Rosmann, Rohan Raavi
```
Comment (tomyyyD) [inline flight/hal/drivers/adm1176.py:7]: damn, what did Rohan do to you

File: flight/tasks/command.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                self.log_info("Switching to NOMINAL state.")
+
+        else:  # Run for all states
+
+            if not DH.data_process_exists("cdh"):
+                DH.register_data_process("cdh", self.data_keys, self.data_format, True, line_limit=500)
+
+            self.log_data[CDH_IDX.TIME] = int(time.time())
+            self.log_data[CDH_IDX.SC_STATE] = SM.current_state
+            self.log_data[CDH_IDX.SD_USAGE] = int(DH.SD_usage() / 1000)  # kb - gets updated in the OBDH task
```
Comment (tomyyyD) [inline flight/tasks/command.py:54]: Need to make sure `SD_usage()` doesn't run very frequently becuase it will take a long time.

**Review by tomyyyD — APPROVED** [review]

File: flight/apps/telemetry/packing.py (hunk 1 above)
Comment (Ibrassow) [reply flight/apps/telemetry/packing.py:40]: No instantiation, the class acts as a namespace for the shared state
Avoid global scope pollution and maintain consistency with the rest of the codebase

File: flight/apps/cdh.py
Comment (Ibrassow) [reply flight/apps/cdh.py:1]: decided will be part of another PR

File: flight/tasks/command.py (hunk 4 above)
Comment (Ibrassow) [reply flight/tasks/command.py:54]: SD_usage only returns the sored class variable from the DataHandler. The actual storage value gets updated in the OBDH task, which for now is commented out (read: other PR with proper cleaning policies to handle this)

File: flight/apps/comms/rf_mcu.py (hunk 2 above)
Comment (Ibrassow) [reply flight/apps/comms/rf_mcu.py:18]: Same comment as above (class variable removed, not needed) + no need for instance behavior (not even shared state this time) and methods will be accessed in other places. The SPI interface will be defined in SATELLITE anyway.

**Pull request merged**